CMakeLists.txt

```diff
@@ -75,7 +70,22 @@ ENDIF()
 
 
 find_program(SED NAMES gsed sed) # gsed - MacOS and FreeBSD, sed - all the rest
-find_file(PCRE_H pcre.h)
+
+pkg_search_module(PCRE REQUIRED libpcre)
+pkg_search_module(JSONC REQUIRED json-c)
+pkg_search_module(LIBGC REQUIRED bdw-gc-threaded bdw-gc)
+pkg_search_module(LIBFFI REQUIRED libffi)
+
+include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ${LIBFFI_INCLUDE_DIRS} /usr/local/include /opt/homebrew/include)
+link_directories(/usr/local/lib /opt/homebrew/lib)
+
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+
+file(READ lib/stdlib.ngs FILE_CONTENTS)
+string(REPLACE "#@INSTALL_LIBDIR@" ${CMAKE_INSTALL_LIBDIR}/ngs FILE_CONTENTS "${FILE_CONTENTS}")
+file(WRITE lib/stdlib.ngs "${FILE_CONTENTS}")
```

Comment:
I would really prefer passing `CMAKE_INSTALL_LIBDIR` to the C code and then exposing it into the language instead of this.